integrations/clip/clip_models.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn as nn
+
+
+class VisualModel(nn.Module):
+    def __init__(self, visual_model, transformations, output_tokens):
+
+        super().__init__()
+
+        self.visual_model = visual_model
+        self.transformations = transformations
+        self.visual_model.output_tokens = output_tokens
+
+    def forward(self, x):
+        return self.visual_model(x)
+
+
+class TextModel(nn.Module):
+    def __init__(
+        self,
+        token_embedding,
+        tokenizer,
+        positional_embedding,
+        transformer,
+        ln_final,
+        text_projection,
+        attn_mask,
+    ):
+
+        super().__init__()
+
+        self.token_embedding = token_embedding
+        self.tokenizer = tokenizer
```

Comment:
where is this tokenizer used?

Reply:
Just for the sample batch. Added to the `TextModel` object for the sake of completeness but not needed in the forward pass. Can remove

Reply:
makes sense, let's slim down any args not needed for onnx export / forward pass